packages/sources/solana-functions/src/shared/buffer-layout-accounts.ts

```diff
@@ -0,0 +1,87 @@
+import { AdapterInputError } from '@chainlink/external-adapter-framework/validation/error'
+import { type Address } from '@solana/addresses'
+import * as BufferLayout from '@solana/buffer-layout'
+import { type Rpc, type SolanaRpcApi } from '@solana/rpc'
+import { MintLayout } from '@solana/spl-token'
+
+interface SanctumPoolState {
+  total_sol_value: bigint
+  trading_protocol_fee_bps: number
+  lp_protocol_fee_bps: number
+  version: number
+  is_disabled: number
+  is_rebalancing: number
+  padding: number[]
+  admin: Uint8Array
+  rebalance_authority: Uint8Array
+  protocol_fee_beneficiary: Uint8Array
+  pricing_program: Uint8Array
+  lp_token_mint: Uint8Array
+}
+
+const SanctumPoolStateLayout = BufferLayout.struct<SanctumPoolState>([
```

Comment:
wouldnt this make more sense to live in the endpoint and be passed as an argument?

seems endpoint specific